py/asmthumb.c

```diff
@@ -90,6 +90,20 @@ void asm_thumb_start_pass(asm_thumb_t *as, uint pass) {
 void asm_thumb_end_pass(asm_thumb_t *as) {
     (void)as;
     // could check labels are resolved...
+#if defined(MCU_SERIES_F7)
+    if (as->pass == ASM_THUMB_PASS_EMIT) {
+        // flush D-cache, so the code emited is stored in memory
+        // Cortex-M7 cache line size is 8 words
+        // Find first and last lines containing new code
+        uintptr_t start = (uintptr_t)as->code_base & (~0x1F);
+        uintptr_t end = (uintptr_t)(as->code_base + as->code_size -1) & (~0x1F);
+        // and flush them
+        SCB_CleanDCache_by_Addr((uint32_t *)start, end - start + 1);
+        
+        // invalidate I-cache
+        SCB_InvalidateICache();
+    }
+#endif
```

Comment:
Should this check to see if the ICache and/or DCache is enabled before making the corresponding call?